### ISSUE TITLE
Doesn't properly sync some messages/statuses posted via slack API

### ISSUE THREAD

**Issue by ldanicel** [hangoutsbot/hangoutsbot#873]
## Description
On my slack account I have a channel called #notifications where all the services are hooked and post notifications.
One of the services is StatusCake.com
Unfortunately, statuscake posts something like in the attached screenshot.
![screenshot_2017-07-24_19-57-14](https://user-images.githubusercontent.com/6535351/28534758-7a969faa-70aa-11e7-920a-18e30f20acc7.png)

## Output of `/bot version`

Bot Version: 3.0.0-beta
Python Version: 3.6.1
* aiohttp 1.3.5
* appdirs 1.4.3
* emoji 0.4.5
* hangups 0.4.2
* telepot 10.5

## Expected behaviour
Hangouts synced message should include all the information in the original message. I don't care if its not pretty formatted like on slack but right now the headline is missing.
## Actual behaviour
Output message on hangouts looks like this:
StatusCake: Your site went down!
Code: 0 - Reason: Timeout / Connection Refused
StatusCake: Your site went down!
Code: 0 - Reason: Timeout / Connection Refused
StatusCake: Your site went back up!
Code: 200 - Downtime: 00:01:11
StatusCake: Your site went back up!
Code: 200 - Downtime: 00:01:10

It is missing the name of the test and the url - I think that is oficially the "title" 
## Steps to reproduce:
 1. Sync slack with hangouts
 2. Set statuscake up and make it send notifications or figure out what json it sends to slack as a notification so that it looks like in the attached screenshot and send notifications like that
## Log output
16:54:18 INFO plugins.slackrtm.bridgeinstance: no hangups user for myslackinstance 
16:54:18 INFO webbridge: slackRTM:receive:{'original_request': {'message': 'Your site went down!\n**Code:** 100 - **Reason:** Timeout / Connection Refused', 'image_id': None, 'segments': None, 'user': 'StatusCake'}, 'chatbridge': {'source_title': 'notifications', 'source_user': 'StatusCake', 'source_uid': '', 'source_gid': '123123213123', 'source_edited': None, 'source_action': None, 'plugin': 'slackRTM'}, 'norelay': ['slackRTM-0']}
16:54:18 INFO webbridge: slackRTM:slackRTM-0:NORELAY:broadcast:['slackRTM-0']
16:54:19 INFO event: eid/dt: 123123123213213-2017-07-24 16:54:16
16:54:19 INFO event: cid/cn: 123123123123/Notifications
16:54:19 INFO event: c/g/un: 123123123123123213123123123/Mybot
16:54:19 INFO event: len/tx: 81/StatusCake: Your site went down!
Code: 100 - Reason: Timeout / Connection Refused
16:54:19 INFO webbridge: slackRTM:slackRTM-0:NORELAY:repeat:['slackRTM-0']

**Comment by OllieTerrance** [comment]
Arguably this is a bug for StatusCake itself: messages with attachments are supposed to carry a suitable fallback for clients that don't support them (such as the IRC/XMPP gateways, and in our case, bridges to other networks).

Looking at that example, given only half the title is linked, I don't think that's in the `title` field of the Slack-side attachment -- the three lines are probably just the text body.  It might be that their fallback doesn't include the first line in the fallback for some reason.

**Comment by ldanicel** [comment]
https://api.slack.com/docs/message-attachments

If you ask me it looks like its the title but...

**Comment by OllieTerrance** [comment]
Slack has `title` and `title_link`, so either the whole of the first line would be linked, or not.

I had a go at mocking up the message [here](https://api.slack.com/docs/messages/builder?msg=%7B%22attachments%22%3A%5B%7B%22fallback%22%3A%22Your%20site%20went%20down!%5Cn*Code%3A*%20100%20-%20*Reason%3A*%20Timeout%20%2F%20Connection%20Refused%22%2C%22color%22%3A%22danger%22%2C%22text%22%3A%22*failure-test%20-%20%3Cranga.ro%3E*%5CnYour%20site%20went%20down!%5Cn*Code%3A*%20100%20-%20*Reason%3A*%20Timeout%20%2F%20Connection%20Refused%22%2C%22mrkdwn_in%22%3A%5B%22text%22%5D%2C%22footer%22%3A%22StatusCake%20Uptime%20Monitoring%22%2C%22footer_icon%22%3A%22https%3A%2F%2Fplatform.slack-edge.com%2Fimg%2Fdefault_application_icon.png%22%7D%5D%7D):

```json
{
    "attachments": [
        {
            "fallback": "Your site went down!\n*Code:* 100 - *Reason:* Timeout / Connection Refused",
            "color": "danger",
            "text": "*failure-test - <ranga.ro>*\nYour site went down!\n*Code:* 100 - *Reason:* Timeout / Connection Refused",
            "mrkdwn_in": ["text"],
            "footer": "StatusCake Uptime Monitoring",
            "footer_icon": "https://platform.slack-edge.com/img/default_application_icon.png"
        }
    ]
}
```

![Preview](https://user-images.githubusercontent.com/4025899/28542502-481df06e-70b4-11e7-9f54-a43cf3fe13a8.png)

If you're able to get the actual message out of the API, that would help to confirm what we're actually dealing with.

**Comment by ldanicel** [comment]
Hey, 
Any update on this ?

Thanks

**Comment by OllieTerrance** [comment]
You are correct that we only take the text -- having taken a proper look, we don't have any handling for attachments at all.  🤔  (Previously I thought we took at least something out of them...)  The SlackRTM plugin is under rewrite at the moment, so I'm unsure what the stance is for adding to the existing plugin.

I've added support in the rewrite as of commit d15fe30; if you want to try it, you'll need to pull from branch [`OllieTerrance-forks:slackrtm-rewrite`](https://github.com/OllieTerrance-forks/hangoutsbot/tree/slackrtm-rewrite), and replace `slackrtm` with `_chatbridge.chatbridge_slackrtm` in your plugins.  But here be dragons: this update still lacks things like join/part sync, and will restructure the `slackrtm` portion of your config (it'll leave a backup of the old plugin's config though).
